src/cmakeTaskProvider.ts

```diff
@@ -161,6 +171,91 @@ export class CMakeTaskProvider implements vscode.TaskProvider {
+                new vscode.CustomExecution(async (resolvedDefinition: vscode.TaskDefinition): Promise<vscode.Pseudoterminal> =>
+                    new CustomBuildTaskTerminal(resolvedDefinition.command, resolvedDefinition.targets, resolvedDefinition.preset, resolvedDefinition.options)
+                ), []);
+            return resolvedTask;
+        }
+        return task;
+    }
+
+    public static async findBuildTask(presetName?: string, targets?: string[]): Promise<CMakeTask | undefined> {
+        // Fetch all CMake task from `tasks.json` files.
+        const allTasks: vscode.Task[] = await vscode.tasks.fetchTasks({ type: CMakeTaskProvider.CMakeScriptType });
+        const tasks: (CMakeTask | undefined)[] = allTasks.map((task: any) => {
+            if (!task.definition.label || !task.group || (task.group && task.group !== vscode.TaskGroup.Build)) {
+                return undefined;
+            }
+            const definition: CMakeTaskDefinition = {
+                type: task.definition.type,
+                label: task.definition.label,
+                command: task.definition.command,
+                targets: task.definition.targets || targets,
+                preset: task.definition.preset,
+                options: task.definition.options
+            };
+
+            const buildTask: CMakeTask = new vscode.Task(definition, vscode.TaskScope.Workspace, task.definition.label, CMakeTaskProvider.CMakeSourceStr);
+            buildTask.detail = task.detail;
+            if (task.group.isDefault) {
+                buildTask.isDefault = true;
+            }
+            return buildTask;
+        });
+
+        const buildTasks: CMakeTask[] = tasks.filter((task) => task !== undefined) as CMakeTask[];
+
+        // No CMake Task is found.
+        if (buildTasks.length === 0) {
+            return undefined;
+        }
+
+        // Find tasks with a target that matches the input preset's target or the input targets
+        let matchingTargetTasks: CMakeTask[];
+
+        if (presetName) {
+            matchingTargetTasks = buildTasks.filter(task => task.definition.preset === presetName);
+        } else {
+            matchingTargetTasks = buildTasks.filter(task => {
+                const taskTargets: string[] = task.definition.targets || [];
+                const inputTargets: string[] = targets || [];
+                return taskTargets.length === inputTargets.length && taskTargets.every((item, index) => item === inputTargets[index]);
+            });
+        }
+
+        if (matchingTargetTasks.length > 0) {
+            // One task is found.
+            if (matchingTargetTasks.length === 1) {
+                return matchingTargetTasks[0];
+            } else {
+                // Search for the matching default task.
+                const defaultTask: CMakeTask[] = matchingTargetTasks.filter(task => task.group?.isDefault);
```

Comment:
You should change `task => task.group?.isDefault` into `task => task.isDefault`
This is because in temporary task created in line 210 you do not add full group but only this one field